OsmAnd/src/net/osmand/plus/onlinerouting/OnlineRoutingHelper.java

```diff
@@ -119,6 +122,18 @@ public String makeRequest(@NonNull String url) throws IOException {
 		return content.toString();
 	}
 
+	public OnlineRoutingEngine createInitStateEngine(RouteCalculationParams params) {
+		String routingProfile = params.mode.getRoutingProfile();
+		boolean isCarBicycleFoot = routingProfile.equals("car") || routingProfile.equals("bicycle") || routingProfile.equals("pedestrian");
```

Comment:
- Method name is very unclear
- Avoid String constants. We have good ApplicationMode which could be tested. 
- Not clear if routing profile should match OSRM Type by vehicle key